internal/controller/operator/factory/build/httproute.go

```diff
@@ -0,0 +1,53 @@
+package build
+
+import (
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+)
+
+// HTTPRoute creates HTTPRoute object
+func HTTPRoute(cr builderOpts, port string, httpRoute *vmv1beta1.EmbeddedHTTPRoute) (*gwapiv1.HTTPRoute, error) {
+	var (
+		defaultPort gwapiv1.PortNumber
+	)
+	if defaultPortVal, err := strconv.Atoi(port); err != nil {
+		return nil, err
+	} else {
+		defaultPort = gwapiv1.PortNumber(defaultPortVal)
+	}
```

Comment:
```suggestion
	defaultPortVal, err := strconv.Atoi(port)
	if err != nil {
		return nil, err
	}
	defaultPort = gwapiv1.PortNumber(defaultPortVal)
```
nitpick: IMO this makes the flow clear